Tests/DocsXsd/DocsXsdTest.php

```diff
@@ -0,0 +1,156 @@
+                'fixtureFile' => 'ValidDocumentationWithAdditionalAttributes.xml',
+            ],
+            'Valid docs example with multiple attributes on the standard element' => [
+                'fixtureFile' => 'ValidStandardWithAdditionalAttributes.xml',
+            ],
+        ];
+    }
+
+    /**
+     * Verify that the invalid XML will throw the correct validation errors
+     *
+     * @dataProvider dataInvalidXsd
+     *
+     * @param string $fixtureFile    The name of the fixture file in the fixture directory.
+     * @param string $expectedStdOut Expected output from the xmllint command.
+     * @param string $expectedStdErr Expected validation error from the xmllint command.
+     *
+     * @return void
+     */
+    public function testInvalidXsd($fixtureFile, $expectedStdOut, $expectedStdErr)
+    {
+        $command = \sprintf(self::COMMAND, $fixtureFile);
+        $result  = $this->executeCliCommand($command);
+
+        $this->assertGreaterThan(0, $result['exitcode'], 'Exit code does not match 0');
+        $this->assertSame($expectedStdOut, $result['stdout'], 'Unexpected output in stdout');
+        $this->assertNotFalse(\strpos($result['stderr'], $expectedStdErr), 'Unexpected output in stderr');
+    }
+
+    /**
+     * Data provider for invalid test cases.
+     *
+     * @return array
+     */
+    public function dataInvalidXsd()
+    {
+        return [
+            'Title attribute too long' => [
+                'fixtureFile'    => 'InvalidTitleTooLong.xml',
+                'expectedStdOut' => '',
+                'expectedStdErr' => "this exceeds the allowed maximum length of '58'",
+            ],
+            'Documentation root element missing' => [
+                'fixtureFile'    => 'InvalidMissingDocumentationRoot.xml',
+                'expectedStdOut' => '',
+                'expectedStdErr' => "Element 'standard': No matching global declaration available for the validation root",
+            ],
+            'Missing title attribute in the documentation root element' => [
+                'fixtureFile'    => 'InvalidMissingTitleAttribute.xml',
+                'expectedStdOut' => '',
+                'expectedStdErr' => "Element 'documentation': The attribute 'title' is required but missing.",
+            ],
+            'Missing standard element in the sequence group' => [
+                'fixtureFile'    => 'InvalidMissingStandardElement.xml',
+                'expectedStdOut' => '',
+                'expectedStdErr' => "Element 'code_comparison': This element is not expected. Expected is ( standard ).",
+            ],
+            'Missing code comparison element in the sequence group' => [
+                'fixtureFile'    => 'InvalidMissingCodeComparisonElement.xml',
+                'expectedStdOut' => '',
+                'expectedStdErr' => "Element 'documentation': Missing child element(s). Expected is ( code_comparison )",
+            ],
```

Comment:
I wonder if this should be invalid.

While the vast majority of Docs will have a standard accompanied by a code comparison, I can imagine situations where a code comparison has no value.

PHPCS also doesn't enforce it in the code in the Generators.

Some examples of docs without code comparisons:
* The docs for the PHPCSDebug sniff in the repo.
* The docs for the PHPCS native `Generic.Files.ByteOrderMark` sniff.
* The docs for the PHPCS native `Generic.Files.EndFileNewline` sniff.
* The docs for the PHPCS native `Generic.Files.EndFileNoNewline` sniff.
* The docs for the PHPCS native `Generic.Files.ExecutableFile` sniff.

I presume there will be more, but these are just _some_ examples.

Reply:
Never really thought about that. Kinda assumed that you need to have some code comparison for the docs to be valid. But I guess I was wrong. In that case, I can add the `minOccurs="0"` attribute to the `code_comparison` type and fix the test case 👍🏼